tests/ui/doc_unsafe.rs

```diff
@@ -88,6 +88,21 @@ very_unsafe!();
 // we don't lint code from external macros
 undocd_unsafe!();
 
+/** This safety section is now recognized.
+
+   # Safety
```

Comment:
The problematic example in the issue has an indentation of 4 spaces. I think this is important because in markdown 4 spaces start a code block, see my other comment.